### PR TITLE
feat: UNIV2-DAI-ETH Collateral

### PR DESCRIPTION
**Pull request by lucas-manuel**
# Description

Spell Description:

COB Team Name or Author(s):

# Contribution Checklist

- [ ] First commit title starts with COB Team Name and Collateral type (e.g. `___-TOKEN-X`)
- [ ] Code approved
- [ ] Tests approved
- [ ] CI Tests pass

# Checklist

- [ ] Every contract variable/method declared as public/external private/internal
- [ ] Verify expiration (`4 days + 2 hours` monthly and `30 days` for the rest)
- [ ] Verify hash in the description matches [here](https://emn178.github.io/online-tools/keccak_256.html)
- [ ] Validate all addresses used are in Kovan changelog
- [ ] Deploy spell to kovan `SOLC_FLAGS="--optimize --optimize-runs=1" dapp --use solc:0.5.12 build --extract && dapp create DssSpell --gas=5000000 --gas-price="$(seth --to-wei 3 "gwei")"`
- [ ] Ensure contract is verified on `kovan` etherscan
- [ ] Change test to use kovan spell address and deploy timestamp
- [ ] Keep `Kovan-DssSpell.sol` and `Kovan-DssSpell.t.sol` the same, but make a copy in `archive`
- [ ] `squash and merge` this PR

### CODE REVIEW

File: src/Kovan-DssSpell.sol [hunk 1]
```diff
@@ -63,14 +68,15 @@ contract SpellAction {
     // Many of the settings that change weekly rely on the rate accumulator
     // described at https://docs.makerdao.com/smart-contract-modules/rates-module
     // To check this yourself, use the following rate calculation (example 8%):
-    //
+    //co
```
Comment (brianmcmichael) [inline src/Kovan-DssSpell.sol:71]: rogue comment

File: src/Kovan-DssSpell.sol [hunk 2]
```diff
@@ -63,14 +68,15 @@ contract SpellAction {
...
     // $ bc -l <<< 'scale=27; e( l(1.08)/(60 * 60 * 24 * 365) )'
     //
     // A table of rates can be found at
     //    https://ipfs.io/ipfs/QmefQMseb3AiTapiAKKexdKHig8wroKuZbmLtPLv4u2YwW
     //
     uint256 constant THREE_PERCENT_RATE = 1000000000937303470807876289;
     uint256 constant SIX_PERCENT_RATE   = 1000000001847694957439350562;
+    uint256 constant ONE_PERCENT_RATE   = 1000000000315522921573372069;
```
Comment (brianmcmichael) [inline src/Kovan-DssSpell.sol:79]: move this above three pct so they are in order

**Comment by brianmcmichael** [comment]
I made a few updates. Getting a revert on the faucet check. We'll need to send some of the mock lp tokens to the faucet. Code should be good otherwise.

File: src/Kovan-DssSpell.t.sol [hunk 3]
```diff
@@ -853,7 +878,76 @@ contract DssSpellTest is DSTest, DSMath {
...
+       spot.poke(ilk);
+
+       // Check faucet amount
+       uint256 faucetAmount = faucet.amt(address(lp));
+       assertTrue(faucetAmount > 0);
+       faucet.gulp(address(lp));
+       assertEq(lp.balanceOf(address(this)), faucetAmount);
+
+       // Check median matches pip.src()
+       assertEq(lpPip.src(), address(lp)); // TODO: Check orbs
```
Comment (brianmcmichael) [inline src/Kovan-DssSpell.t.sol:904]: This is failing. Is the oracle configured correctly?

```
  «Error: Wrong `address' value»
    Expected: 0xfd5608515a47c37afba68960c1916b79af9491d0
      Actual: 0xb10cf58e08b94480fcb81d341a63295ebb2062c2
```

Comment (brianmcmichael) [reply src/Kovan-DssSpell.t.sol:904]: Oh....

The Kovan Oracle is using the actual UniV2 LP token on-chain, but we have a dummy token in it's place. This complicates things.

**Comment by brianmcmichael** [comment]
We'll need to back this up and redeploy flips and joins but use the real LP exchange for this one. LPL can generate LP tokens at uniswap on Kovan if they need to test and we can remove faucet check from the integration test for this one.

So, flip and join will change, and we'll use [0xb10cf58e08b94480fcb81d341a63295ebb2062c2](https://kovan.etherscan.io/address/0xb10cf58e08b94480fcb81d341a63295ebb2062c2) as the token.

File: src/Kovan-DssSpell.t.sol (hunk 3 above)
Comment (gbalabasquer) [reply src/Kovan-DssSpell.t.sol:904]: I think we should use the real tokens in this case. We can generate DAI and we have plenty kovan ETH, so maybe this shouldn't be a dummy token at all, nor be in the faucet.

Comment (gbalabasquer) [reply src/Kovan-DssSpell.t.sol:904]: Oops, I should have read your next comment before writing.

**Pull request merged**